source/NVDAObjects/behaviors.py

```diff
@@ -265,15 +265,34 @@ def event_textChange(self):
 		"""
 		self._event.set()
 
-	def _getTextLines(self):
-		"""Retrieve the text of this object in lines.
+	def _get_diffAlgo(self):
+		return (
+			diffHandler.dmp
+			if config.conf["terminals"]["diffAlgo"] == "dmp"
+			else diffHandler.difflib
+		)
+
+	def _get_devInfo(self):
+		info = super().devInfo
+		info.append(f"diffing algorithm: {self.diffAlgo}")
+		return info
+
+	def _getText(self) -> str:
+		"""Retrieve the text of this object.
 		This will be used to determine the new text to speak.
 		The base implementation uses the L{TextInfo}.
 		However, subclasses should override this if there is a better way to retrieve the text.
-		@return: The current lines of text.
-		@rtype: list of str
 		"""
-		return list(self.makeTextInfo(textInfos.POSITION_ALL).getTextInChunks(textInfos.UNIT_LINE))
+		if hasattr(self, "_getTextLines"):
+			log.warning("LiveText._getTextLines is deprecated, please override _getText instead.")
+			return '\n'.join(self._getTextLines())
+		ti = self.makeTextInfo(textInfos.POSITION_ALL)
+		if self.diffAlgo.unit == textInfos.UNIT_CHARACTER:
+			return ti.text
+		elif self.diffAlgo.unit == textInfos.UNIT_LINE:
+			return "\n".join(ti.getTextInChunks(textInfos.UNIT_LINE))
+		else:
+			raise NotImplementedError(f"Unknown unit {self.diffAlgo.unit}")

```

Comment:
Given this is the only place that unit is used, This is another if statement that can be replaced with polymorphic behavior:
- Add a an abstract `getText` method to the DiffAlgo base class.
- Do each of the above in each implementation.

There is no need for the final else and exception raising.

Reply:
If it isn't clear, getText will need the signature: `getText(ti: TextInfo) -> str`

Reply:
It is also used on the `__repr__` implementation. Should I hardcode a name?

Reply:
If you are talking about `DiffMatchPatch.__repr__` and `Difflib.__repr__`, those implementations could just be internal to each of the classes. There doesn't need to be a base implementation of `__repr__` because it is valid on all objects.

I'd also get rid of `name`, the class name is much more useful for a developer. You can look at `__qualname__` which is pretty useful in situations like this.

This will just leave you with `diff`, which is fine. It would be useful if the class docstring for each impl mentioned something like: "This is a character/line based diff, which means xyz"